packages/widgets/user-profile-widget/src/lib/widget/mixins/initMixin/initComponentsMixins/initUserCustomAttributesMixin.ts

```diff
@@ -0,0 +1,169 @@
+import { initWidgetRootMixin } from './initWidgetRootMixin';
+
+export const initUserCustomAttributesMixin = createSingletonMixin(
+  <T extends CustomElementConstructor>(superclass: T) =>
+    class UserCustomAttributesMixinClass extends compose(
+      stateManagementMixin,
+      loggerMixin,
+      initWidgetRootMixin,
+      modalMixin,
+    )(superclass) {
+      // flow Id is key in all maps
+      #editModals: Record<string, ModalDriver> = {};
+
+      #editFlows: Record<string, FlowDriver> = {};
+
+      #deleteModals: Record<string, ModalDriver> = {};
+
+      #deleteFlows: Record<string, FlowDriver> = {};
+
+      #initEditModalContent(flowId: string) {
+        this.#editModals[flowId]?.setContent(
+          createFlowTemplate({
+            projectId: this.projectId,
+            flowId,
+            baseUrl: this.baseUrl,
+            baseStaticUrl: this.baseStaticUrl,
+          }),
+        );
+        this.#editFlows[flowId]?.onSuccess(() => {
+          this.#editModals[flowId]?.close();
+          this.actions.getMe();
+        });
+      }
+
+      // have 2 init functions for edit and delete modals in order to keep the same standards as the email/phone/name mixin
+      #initDeleteModalContent(flowId: string) {
+        this.#deleteModals[flowId]?.setContent(
+          createFlowTemplate({
+            projectId: this.projectId,
+            flowId,
+            baseUrl: this.baseUrl,
+            baseStaticUrl: this.baseStaticUrl,
+          }),
+        );
+        this.#deleteFlows[flowId]?.onSuccess(() => {
+          this.#deleteModals[flowId]?.close();
+          this.actions.getMe();
+        });
+      }
+
+      #updateCustomValueUserAttrs = withMemCache(
+        (userCustomAttributes: ReturnType<typeof getUserCustomAttrs>) => {
+          const allCustomAttributesComponents =
+            this.shadowRoot?.querySelectorAll(
+              'descope-user-attribute[data-id^="customAttributes."]',
+            );
+
+          Array.from(allCustomAttributesComponents).forEach((nodeEle) => {
+            const attrName = nodeEle.getAttribute('data-id');
+            const customAttrName = attrName.replace('customAttributes.', '');
+            const type =
+              nodeEle.getAttribute('data-type') || AttributeTypeName.TEXT;
+            const val = userCustomAttributes[customAttrName];
+
+            const compInstance = new UserAttributeDriver(nodeEle, {
+              logger: this.logger,
+            });
+
+            if (type === AttributeTypeName.DATE && val) {
+              // to full date time
+              compInstance.value = new Date(val).toLocaleString();
+            } else if (
+              type === AttributeTypeName.BOOLEAN &&
+              val !== undefined
+            ) {
+              compInstance.value = !val ? 'False' : 'True';
+            } else {
+              compInstance.value = (val || '').toString();
+            }
```

Comment:
Consider extrcting this logic into a separate fn
`getFormattedValue(val, type) => string`